src/confdb/data/ReleaseVersionInfo.java

```diff
@@ -0,0 +1,132 @@
+            major = Integer.parseInt(splitTag[2]);
+            if (splitTag[3].equals("X")){ //for integration builds
+                minor = -1;
+            }else{
+                minor = Integer.parseInt(splitTag[3]);
+            }
+        } catch (NumberFormatException ex) {            
+            return false;            
+        }
+
+        if(splitTag.length>4){            
+            if(splitTag[4].startsWith("pre")){
+                try{                    
+                    prerel = Integer.parseInt(splitTag[4].substring(3));
+                } catch(NumberFormatException ex){
+
+                }
+            }
+            if(splitTag[4].startsWith("patch")){
+                try{
+                    patch = Integer.parseInt(splitTag[4].substring(5));
+                } catch(NumberFormatException ex){
+                    
+                }
+            }
+        }
+        //System.err.print("parsed "+releaseTag);
+        //System.err.println("  "+cycle+" "+major+" "+minor+" pre "+prerel+" patch "+patch+" ");
+        return true;
+
+    }
+    public Integer cycle(){return cycle;}
+    public Integer major(){return major;}
+    public Integer minor(){return minor;} 
+    public Integer patch(){return patch;} 
+    public Integer prerel(){return prerel;} 
+    /** 
+     * checks if a given release is >= the specified values
+     * 
+     * notes:
+     *  a prerel is below its minor version
+     *  a patch is above its minor version
+     *  a nightly (X) is undefined, defaults to above as have to do something  
+     */
+    public boolean geq(int cycle,int major,int minor,int prerel,int patch){
+        if(this.cycle > cycle) return true;
+        else if(this.cycle==cycle && this.major > major) return true;
+        else if(this.cycle==cycle && this.major == major) {
+            if(minor==-1 || this.minor>minor) return true;
+            else if(this.minor==minor){
+                if(prerel!=-1){
+                    return this.prerel==null || this.prerel >= prerel;                    
+                }else if(patch!=-1){
+                    return this.patch!=null && this.patch>=patch;
+                }else{
+                    return true;
+                }
+            }
+            else return false;
+        }
+        return false;
+    }
+    public boolean geq(int cycle,int major,int minor){
+        return geq(cycle,major,minor,-1,-1);
+    }          
+    public boolean geq(int cycle,int major,int minor,int prerel){
+        return geq(cycle,major,minor,prerel,-1);
+    }          
+
+    public boolean equals(int cycle,int major,int minor,int prerel,int patch){
+        if(this.cycle==cycle && this.major==major && this.minor==minor){
+            if(prerel==-1 && this.prerel==null){
+                if(patch==-1 && this.patch==null) return true;
+                else return this.patch!=null && this.patch==patch;
+            } return this.prerel!=null && this.prerel==prerel;
```

Comment:
```suggestion
            } else return this.prerel!=null && this.prerel==prerel;
```
Or maybe just something like.. ?
```java
int thisPrerel = this.prerel==null ? -1 : this.prerel.intValue();
int thisPatch = this.patch==null ? -1 : this.patch.intValue();
return (this.cycle==cycle && this.major==major && this.minor==minor && thisPrerel==prerel && thisPatch==patch);
```